File: EasyTab/EasyTab.Services/Database/UserRole.cs
using EasyTab.Services.Interfaces;
using System;
using System.Collections.Generic;

namespace EasyTab.Services.Database;

public partial class UserRole : ISoftDelete
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public int RoleId { get; set; }

    public bool IsDeleted { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual Role Role { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: EasyTab/UI/easytab_mobile/lib/models/userrole.dart
import 'package:easytab_mobile/models/role.dart';
import 'package:easytab_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'userrole.g.dart';

@JsonSerializable()
class UserRole {
  int? id;
  int? roleId;
  int? userId;
  Role? role;
  User? user;
  bool? isDeleted;
  DateTime? deletedAt;

  UserRole({
    this.id,
    this.roleId,
    this.userId,
    this.role,
    this.user,
    this.isDeleted,
    this.deletedAt,
  });

  factory UserRole.fromJson(Map<String, dynamic> json) =>
      _$UserRoleFromJson(json);
  Map<String, dynamic> toJson() => _$UserRoleToJson(this);
}
